addOns/client/src/main/java/org/zaproxy/addon/client/ui/PopupMenuExportClientMap.java

```diff
@@ -0,0 +1,290 @@
+                throws IOException {
+            // We could create a set of data structures and use jackson, but the format is very
+            // simple and this is much more memory efficient - it still uses jackson for value
+            // output
+            String indent = " ".repeat(level * 2);
+
+            outputKV(fw, indent, true, NODE_KEY, level == 0 ? ROOT_NODE_NAME : node.toString());
+
+            outputKV(fw, indent, false, NAME_KEY, node.getUserObject().getName());
+            outputKV(fw, indent, false, URL_KEY, node.getUserObject().getUrl());
+            outputKV(fw, indent, false, STORAGE_KEY, node.getUserObject().isStorage());
+            outputKV(fw, indent, false, VISITED_KEY, node.getUserObject().isVisited());
+            if (node.getUserObject().getComponents() != null
+                    && !node.getUserObject().getComponents().isEmpty()) {
+                fw.write(indent + "  ");
+                fw.write("components:\n");
+
+                boolean first = true;
+                for (ClientSideComponent component : node.getUserObject().getComponents()) {
+                    first = outputKVIfNotEmpty(fw, indent, first, "href", component.getHref());
+                    first = outputKVIfNotEmpty(fw, indent, first, "id", component.getId());
+                    first =
+                            outputKVIfNotEmpty(
+                                    fw, indent, first, "tagName", component.getTagName());
+                    first =
+                            outputKVIfNotEmpty(
+                                    fw, indent, first, "tagType", component.getTagType());
+                    first = outputKVIfNotEmpty(fw, indent, first, "text", component.getText());
+                    first = outputKVIfNotEmpty(fw, indent, first, "type", component.getType());
+                    first =
+                            outputKVIfNotEmpty(
+                                    fw,
+                                    indent,
+                                    first,
+                                    "typeForDisplay",
+                                    component.getTypeForDisplay());
+                    first = outputKVIfNotEmpty(fw, indent, first, "formId", component.getFormId());
+                    first =
+                            outputKVIfNotEmpty(
+                                    fw,
+                                    indent,
+                                    first,
+                                    "isStorageEvent",
+                                    component.isStorageEvent());
+                    first = true;
+                }
+            }
+
+            Stats.incCounter(STATS_EXPORT_CLIENTMAP + ".node");
+
+            if (node.getChildCount() > 0) {
+                fw.write(indent);
+                fw.write("  ");
+                fw.write(CHILDREN_KEY);
+                fw.write(": ");
+                fw.newLine();
+                node.children()
```

Comment:
The children should be ordered, otherwise we cant use them when comparing Client Maps.

Reply:
The tree is sorted already?

Reply:
It doesnt look like the components are, at least not from your example output

Reply:
Okay, will dig into it.

Reply:
Children are sorted:

```
 C:\Users\user\Desktop\fred.yaml (54 hits)
	Line    1: - node: "ClientMap"
	Line    7:   - node: "https://github.com/"
	Line   13:     - node: "https://github.com/OWASP/"
	Line   19:       - node: "https://github.com/OWASP/owasp-swag/"
	Line   25:         - node: "https://github.com/OWASP/owasp-swag/tree/"
	Line   31:           - node: "https://github.com/OWASP/owasp-swag/tree/master/"
	Line   37:             - node: "https://github.com/OWASP/owasp-swag/tree/master/projects/"
	Line   43:               - node: "https://github.com/OWASP/owasp-swag/tree/master/projects/juice-shop"
	Line   48:   - node: "https://juice-shop.herokuapp.com/"
	Line   54:     - node: "https://juice-shop.herokuapp.com/#"
	Line   60:       - node: "https://juice-shop.herokuapp.com/#!"
	Line   65:       - node: "https://juice-shop.herokuapp.com/#/"
	Line  252:       - node: "https://juice-shop.herokuapp.com/#/about"
	Line  465:       - node: "https://juice-shop.herokuapp.com/#/contact"
	Line  564:       - node: "https://juice-shop.herokuapp.com/#/forgot-password"
	Line  569:       - node: "https://juice-shop.herokuapp.com/#/login"
	Line  697:       - node: "https://juice-shop.herokuapp.com/#/photo-wall"
	Line  832:       - node: "https://juice-shop.herokuapp.com/#/register"
	Line  950:       - node: "https://juice-shop.herokuapp.com/#/score-board"
	Line  955:       - node: "https://juice-shop.herokuapp.com/#/search"
	Line 1034:     - node: "https://juice-shop.herokuapp.com/ftp/"
	Line 1040:       - node: "https://juice-shop.herokuapp.com/ftp/legal.md"
	Line 1045:     - node: "https://juice-shop.herokuapp.com/redirect?to=https://github.com/juice-shop/juice-shop"
	Line 1050:     - node: "https://juice-shop.herokuapp.com/Cookies"
	Line 1074:   - node: "https://opensea.io/"
	Line 1080:     - node: "https://opensea.io/collection/"
	Line 1086:       - node: "https://opensea.io/collection/juice-shop"
	Line 1091:   - node: "https://owasp-juice.shop/"
	Line 1097:     - node: "https://owasp-juice.shop/"
	Line 1102:   - node: "https://owasp.org/"
	Line 1108:     - node: "https://owasp.org/"
	Line 1113:     - node: "https://owasp.org/slack/"
	Line 1119:       - node: "https://owasp.org/slack/invite"
	Line 1124:     - node: "https://owasp.org/www-project-juice-shop/"
	Line 1130:       - node: "https://owasp.org/www-project-juice-shop/#"
	Line 1136:         - node: "https://owasp.org/www-project-juice-shop/#div-userday2024"
	Line 1141:   - node: "https://twitter.com/"
	Line 1147:     - node: "https://twitter.com/intent/"
	Line 1153:       - node: "https://twitter.com/intent/tweet?text=Magn(et)ificent!%20(%C2%A9%20bkimminich)%20@owasp_juiceshop&hashtags=appsec"
	Line 1158:       - node: "https://twitter.com/intent/tweet?text=%F0%9F%98%BC%20"
	Line 1164:         - node: "https://twitter.com/intent/tweet?text=%F0%9F%98%BC%20#"
	Line 1170:           - node: "https://twitter.com/intent/tweet?text=Welcome%20to%20the%20Bee%20Haven%20(/#/bee-haven)%F0%9F%90%9D%20(%C2%A9%20evmrox)%20@owasp_juiceshop&hashtags=appsec"
	Line 1175:           - node: "https://twitter.com/intent/tweet?text=%F0%9F%98%BC%20#zatschi%20#whoneedsfourlegs%20@owasp_juiceshop&hashtags=appsec"
	Line 1180:     - node: "https://twitter.com/owasp_juiceshop"
	Line 1185:   - node: "https://www.eventbrite.com/"
	Line 1191:     - node: "https://www.eventbrite.com/e/"
	Line 1197:       - node: "https://www.eventbrite.com/e/owasp-global-appsec-san-francisco-2024-tickets-723699172707"
	Line 1202:   - node: "https://www.facebook.com/"
	Line 1208:     - node: "https://www.facebook.com/owasp.juiceshop"
	Line 1213:   - node: "https://www.reddit.com/"
	Line 1219:     - node: "https://www.reddit.com/r/"
	Line 1225:       - node: "https://www.reddit.com/r/owasp_juiceshop"
	Line 1230:   - node: "https://www.youtube.com/"
	Line 1236:     - node: "https://www.youtube.com/watch?v=9PnbKL3wuH4"
```

Components are not, but components don't have a consistent set of fields (which I brought up a number of weeks ago).